iw/filter_gui.py

```diff
@@ -130,6 +130,24 @@ def __init__(self, xml):
     def addToUI(self, tuple):
         pass
 
+    def incrementDeviceCount(self, tuple):
+        global totalDevices, totalSize
+        global selectedDevices, selectedSize
+        global talliedDevices
+
+        # Check to be sure this device has not already been counted
+        if tuple[0]["name"] in talliedDevices or tuple[0]["DEVNAME"] in talliedDevices:
+            return
+
+        talliedDevices.append(tuple[0]["name"])
+
+        totalDevices += 1
+        totalSize += tuple[0]["XXX_SIZE"]
+
+        if tuple[2]:
+            selectedDevices += 1
+            selectedSize += tuple[0]["XXX_SIZE"]
```

Comment:
I think these methods could be made clearer if the tuples were expanded somewhere. ``tuple[X]`` is really cryptic.